src/core.c/Stash.pm6

```diff
@@ -62,15 +71,31 @@ my class Stash { # declared in BOOTSTRAP
         nextwith(key.Str, assignval)
     }
 
+    proto method BIND-KEY(|) {*}
```

Comment:
OOC, why the proto?

Reply:
Hmmm...  that feels like a premature optimization to me...  I mean, if the performance issue were real, there are many other cases in the core we would need to look at.  And I don't think it was ever an issue really, apart from really cold code.  And even less so since new-disp?  @jnthn @niner

Reply:
Could be. That's why I used 'possibly'. But in either case, the two candidates provided by the class are all it needs. So, `proto` makes sense even from this perspective.

I'd rather ask why does it worry you?

Reply:
Well, it is a bit like (lack of) action at a distance.  Any subclass of `Stash` will *not* have access to the `Hash` candidates, like ever.  I think so far, additional `proto`s in the setting, have been used only to fix potential ambiguities.

Reply:
If you use the `proto`s here, I would add a little comment as to *why* they were added, so that our future self / others might find it easier to decide on their need if the situation changes.

Reply:
`Hash` variants are thread-unsafe. Therefore any `Stash`-descendant has to well justify their use. Either way, `Stash` overrides them completely making `self.Hash::BIND-KEY` the only way to address them.

I'll try to remember about the comment. Makes sense.